filebeat/reader/multiline/multiline.go

```diff
@@ -262,13 +264,23 @@ func (mlr *Reader) clear() {
 	mlr.message = reader.Message{}
 	mlr.last = nil
 	mlr.numLines = 0
+	mlr.truncated = 0
 	mlr.err = nil
 }
 
 // finalize writes the existing content into the returned message and resets all reader variables.
 func (mlr *Reader) finalize() reader.Message {
+	if mlr.truncated > 0 {
+		mlr.message.AddFields(common.MapStr{
+			"event": common.MapStr{
+				"truncated": true,
+			},
+		})
+	}
+
```

Comment:
The `mlr.truncate > 0` branch do in the worst case 4 allocations on every run.
I would define the _common.MapStr_ in a package variable that I would reuse.